code/unclassified/src/apriori/apriori.py

```diff
@@ -0,0 +1,131 @@
+# Part of Cosmos by OpenGenus Foundation
+# Find frequently occuring subsets of items in a set of transactions
+# Contributed by: Aditya Lohana (dundermiflin)
+
+
+import csv
+from itertools import combinations, chain
+
+
+min_support = 0.1
+
+
+def csv_to_list(filename):
+	"""
+		Function to convert the transaction-dataset into a list of lists
+	"""
+	with open(filename, 'r') as f:
+		reader = csv.reader(f)
+		data = [row for row in reader]
+		return data
+
+
+def count_support(transactions):
+	"""
+		Function to count the support of each unique item in the dataset
+		Returns a dictionary with item names as keys, and support as values
+	"""
+	sup = {}
+	num_transactions = len(transactions)
+	for row in transactions:
+		for col in row:
+			if col not in sup.keys():
+				sup[col] = 1
+			else:
```

Comment:
I notice what you're trying to do is create a sort of counter object. I recommend you use a `defaultdict` instead and convert it to a dict all the way at the end.

Reply:
Nice catch! Had no idea something like this existed. Made the changes.